isamples_metadata/GEOMETransformer.py

```diff
@@ -0,0 +1,362 @@
+        order = self._source_record_main_record().get("order")
+        if order is not None:
+            keywords.append(order)
+        phylum = self._source_record_main_record().get("phylum")
+        if phylum is not None:
+            keywords.append(phylum)
+        family = self._source_record_main_record().get("family")
+        if family is not None:
+            keywords.append(family)
+        classname = self._source_record_main_record().get("class")
+        if classname is not None:
+            keywords.append(classname)
+        return keywords
+
+    def produced_by_id_string(self) -> typing.AnyStr:
+        parent_record = self._source_record_parent_record()
+        if parent_record is not None:
+            return parent_record["bcid"]
+        return Transformer.NOT_PROVIDED
+
+    def produced_by_label(self) -> typing.AnyStr:
+        parent_record = self._source_record_parent_record()
+        if parent_record is not None:
+            label_pieces = []
+            event_id = parent_record.get("eventID")
+            if event_id is not None:
+                label_pieces.append(event_id)
+            expedition_code = parent_record.get("expeditionCode")
+            if expedition_code is not None:
+                label_pieces.append(expedition_code)
+            return " ".join(label_pieces)
+        return Transformer.NOT_PROVIDED
+
+    def produced_by_description(self) -> typing.AnyStr:
+        parent_record = self._source_record_parent_record()
+        if parent_record is not None:
+            description_pieces = []
+            event_remarks = parent_record.get("eventRemarks")
+            if event_remarks is not None:
+                description_pieces.append(event_remarks)
+            expedition_code = parent_record.get("expeditionCode")
+            if expedition_code is not None:
+                description_pieces.append(f"expeditionCode: {expedition_code}")
+            sampling_protocol = parent_record.get("samplingProtocol")
+            if sampling_protocol is not None:
+                description_pieces.append(f"samplingProtocol: {sampling_protocol}")
+            permit_information = parent_record.get("permitInformation")
+            if permit_information is not None:
+                description_pieces.append(f"permitInformation: {permit_information}")
+            taxonomy_team = parent_record.get("taxTeam")
+            if taxonomy_team is not None:
+                description_pieces.append(f"taxonomy team: {taxonomy_team}")
+            project_id = parent_record.get("projectId")
+            if project_id is not None:
+                description_pieces.append(f"projectId: {project_id}")
+            return " | ".join(description_pieces)
+        return Transformer.NOT_PROVIDED
+
+    def produced_by_feature_of_interest(self) -> typing.AnyStr:
+        # TODO: implement
+        # "[infer from specimen category, locality; need to so some unique values analysis]"
```

Comment:
Again, this comment was mostly skipped as we need more definition here.  One of the test examples used the microhabitat so I included it.